ui/ui-components/components/record/SampleRecordCard.tsx

```diff
@@ -0,0 +1,357 @@
+  );
+
+  const saveRecord = useCallback(
+    (record, groups = undefined, destinations = undefined) => {
+      setLoading(true);
+      setRecord(record);
+      setGroups(groups);
+      setDestinations(destinations);
+      setRecordId(record.id);
+      setHasRecords(true);
+      setLoading(false);
+
+      setCachedSampleRecordId(model.id, record.id);
+    },
+    [model]
+  );
+
+  const loadRecord = useCallback(async () => {
+    setLoading(true);
+
+    let record: Models.GrouparooRecordType;
+    let groups: Models.GroupType[];
+    let destinations: Models.DestinationType[];
+
+    if (recordId) {
+      ({ record, groups, destinations } = await execApi<Actions.RecordView>(
+        "get",
+        `/record/${recordId}`
+      ));
+    } else {
+      ({ record, groups, destinations } = await execApi<Actions.RecordsList>(
+        "get",
+        "/records",
+        {
+          limit: 25,
+          offset: 0,
+          modelId: model.id,
+        }
+      ).then((response) => {
+        if (response.total === 0) {
+          setHasRecords(false);
+        } else {
+          const randomRecord = response?.records?.length
+            ? response.records[
+                Math.floor(Math.random() * response.records.length)
+              ]
+            : undefined;
+
+          if (randomRecord?.id) {
+            return execApi<Actions.RecordView>(
+              "get",
+              `/record/${randomRecord.id}`
+            );
+          }
+        }
+
+        return {
+          record: undefined,
+          groups: undefined,
+          destinations: undefined,
+        };
+      }));
+    }
+
+    if (record?.id) {
+      saveRecord(record, groups, destinations);
+    } else {
+      // Got an invalid record. Let's clear this and start over.
+      clearCachedSampleRecordId(model.id);
+      setRecordId(undefined);
+    }
+
+    setLoading(false);
+  }, [recordId, saveRecord, execApi, model]);
+
+  useEffect(() => {
+    // Switched to another model
+    if (prevModelId && prevModelId !== model.id) {
+      setLoading(true);
+      setRecordId(getCachedSampleRecordId(model.id));
+      setRecord(undefined);
+      setGroups(undefined);
+      setDestinations(undefined);
+      setHasRecords(true);
+      setLoading(false);
+    }
+  }, [record, model]);
+
+  useEffect(() => {
+    if (!loading && hasRecords && (!record || record.id !== recordId)) {
+      loadRecord();
+    }
+  }, [record, recordId, loading, loadRecord, hasRecords]);
+
+  const importRecord = async () => {
+    setImporting(true);
+    const response = await execApi<Actions.RecordImport>(
+      "post",
+      `/record/${record.id}/import`
+    );
+    if (response?.record) {
+      setRecord(response.record);
+      setGroups(response.groups);
+      successHandler.set({ message: "Import Complete!" });
+    } else {
+      loadRecord(); // we may have done a partial import
+    }
+    setImporting(false);
+  };
+
+  const actions = [
+    <LoadingButton
```

Comment:
React says these need `key`'s

Reply:
This is just an array of elements that do eventually get assigned a key. No errors in the console for this.